spec/services/preserved_object_handler_create_spec.rb

```diff
@@ -2,49 +2,33 @@
 require 'services/shared_examples_preserved_object_handler'
 
 RSpec.describe PreservedObjectHandler do
-  before do
-    allow(Dor::WorkflowService).to receive(:update_workflow_error_status)
-  end
-
   let(:druid) { 'ab123cd4567' }
   let(:incoming_version) { 6 }
   let(:incoming_size) { 9876 }
   let(:storage_dir) { 'spec/fixtures/storage_root01/sdr2objects' }
   let(:ms_root) { MoabStorageRoot.find_by(storage_location: storage_dir) }
   let(:po_handler) { described_class.new(druid, incoming_version, incoming_size, ms_root) }
   let(:exp_msg) { "added object to db as it did not exist" }
-  let(:po_args) do
-    {
-      druid: druid,
-      current_version: incoming_version,
-      preservation_policy_id: PreservationPolicy.default_policy.id
-    }
-  end
-  let(:cm_args) do
-    {
-      preserved_object: an_instance_of(PreservedObject), # TODO: ensure we got the preserved object that we expected
-      version: incoming_version,
-      size: incoming_size,
-      moab_storage_root: ms_root,
-      status: 'validity_unknown' # NOTE: ensuring this particular status is the default
-      # NOTE: lack of validation timestamps here
-    }
-  end
+
+  before { allow(Dor::WorkflowService).to receive(:update_workflow_error_status) }
 
   describe '#create' do
     it 'creates PreservedObject and CompleteMoab in database' do
-      expect(PreservedObject).to receive(:create!).with(po_args).and_call_original
-      expect(CompleteMoab).to receive(:create!).with(cm_args).and_call_original
       po_handler.create
+      new_po = PreservedObject.find_by(druid: druid)
+      new_cm = new_po.complete_moabs.find_by(version: incoming_version)
+      expect(new_po.current_version).to eq incoming_version
+      expect(new_cm.moab_storage_root).to eq ms_root
+      expect(new_cm.size).to eq incoming_size
```

Comment:
i'd be inclined to confirm `status` is `validity_unknown` on `new_cm`.  would also prob check for `nil` validation timestamps (per the comment in the old `cm_args` hash).

Reply:
I believe that is covered in other examples.  I think of this example being about the critical **identity** of the created objects rather than the complete examination of every attribute.  YMMV.

Reply:
the next test does check those attributes, but it's for `po_handler.create(true)`, whereas this one's about `po_handler.create`, which i'd expect to create objects with differing values for status and validation times.